### PR TITLE
[Dialogs] Customize tint color of alert title icon

### PR DESCRIPTION
**Pull request by galiak11**
Update AlertController API to customize the alert title, including: title alignment, title icon & titleIcon tint color.

#5174
[b/116100167](http://b/116100167)
b/113257098
b/113257097

### CODE REVIEW

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 1]
```diff
@@ -194,6 +195,19 @@ - (void)setTitleIcon:(UIImage *)titleIcon {
...
 }
 
+- (void)setTitleIconTintColor:(UIColor *)titleIconTintColor {
+  _titleIconTintColor = titleIconTintColor;
+  [self tintTitleIcon];
+}
+
+- (void)tintTitleIcon {
+  if (self.titleIconTintColor != nil && self.titleIcon != nil) {
+    UIImage *image = [self.titleIcon imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
```
Comment (ianegordon) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: Typically I would not expect us to re-initialize this image.  I would expect the client to give us an image with the proper rendering mode.
Based on that, can we eliminate this method completely?
In setTitleIcon: and setTitleIconTintColor: instead of [self tintTitleIcon], just call self.titleIconImageView.tint = tint.

**Review by ianegordon — DISMISSED** [review]

File: components/ActionSheet/tests/unit/MDCActionSheetTableCellTest.m [hunk 2]
```diff
@@ -139,7 +139,7 @@ - (void)testSetActionFont {
   // When
   self.actionSheet.actionFont = actionFont;
   NSArray *cells = [MDCActionSheetTestHelper getCellsFromActionSheet:self.actionSheet];
-  XCTAssertNotEqual(cells.count, 0);
+  XCTAssertNotEqual(cells.count, 0.0f);
```
Comment (ianegordon) [inline components/ActionSheet/tests/unit/MDCActionSheetTableCellTest.m]: We should not be comparing an NSUInteger with a floating point value.

This seems unrelated to dialogs, did a stray commit get pulled in?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 3]
```diff
@@ -198,6 +202,11 @@ - (void)setTitleIcon:(UIImage *)titleIcon {
   [self.titleIconImageView sizeToFit];
 }
 
+- (void)setTitleIconTintColor:(UIColor *)titleIconTintColor {
+  _titleIconTintColor = titleIconTintColor;
+  self.titleIconImageView.tintColor = titleIconTintColor;
```
Comment (ianegordon) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m:208]: Since self.titleIconImageView doesn't always exist, it might be worth wrapping this in an if to explicitly communicate this with engineers.
if (self.titleIconImageView) {
  self.titleIconImageView.tintColor = titleIconTintColor;
}

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 4]
```diff
@@ -198,6 +202,11 @@ - (void)setTitleIcon:(UIImage *)titleIcon {
   [self.titleIconImageView sizeToFit];
 }
 
+- (void)setTitleIconTintColor:(UIColor *)titleIconTintColor {
+  _titleIconTintColor = titleIconTintColor;
```
Comment (ianegordon) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m:207]: You will need to apply _titleIconTintColor to newly created titleIconImageView.
Add this in setTitleIcon:

**Review by ianegordon — APPROVED** [review]
Please resolve issues and get successful kokoro builds before merge.

File: components/ActionSheet/tests/unit/MDCActionSheetTableCellTest.m (hunk 2 above)
Comment (galiak11) [reply components/ActionSheet/tests/unit/MDCActionSheetTableCellTest.m]: It was suppose to be an unsigned int. either way - this is already fixed in develop, so im removing this from the commit.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 4 above)
Comment (galiak11) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m:207]: Good catch. Done.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 3 above)
Comment (galiak11) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m:208]: Done.

**Comment by galiak11** [comment]
Updating AlertController API to customize the tintColor of the alert's titleIcon.

Includes tests and examples.
issue: #5174
issue: Update MDC API to customize MDCAlertController title🐛
